demos/TransferLearning/TransferLearningGui.py

```diff
@@ -0,0 +1,219 @@
+import os
+import random
+
+import ipywidgets as widgets
+import subprocess
+import TransferLearningDemo
+
+import tensorflow as tf
+import numpy as np
+
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+
+from IPython.display import display
+from IPython.display import clear_output
+
+ngraph_backends = ['PLAIDML', 'CPU', 'DISABLED']
+batch_size = 16
+epochs = 5
+backend = ngraph_backends[0]
+
+
+class ProgressBar(tf.keras.callbacks.Callback):
+
+    def __init__(self, demo):
+        self.epoch = 0
+        self.train_steps_per_epoch = demo.train_generator.n // demo.batch_size
+
+    def on_train_begin(self, logs=None):
+        self.total_epochs = epoch_slider.value
+        progress_bar.value = progress_bar.min
+        progress_bar.max = self.train_steps_per_epoch * self.total_epochs
+        epoch_text = ""
+        #self.demo.progress_text.value = "Training"
+        #self.train_value = 0
+
+    def on_train_end(self, logs=None):
+        #self.demo.progress_bar.value = self.demo.progress_bar.max
+        #self.demo.progress_text.value = "Training Complete"
+        pass
+
+    def on_epoch_begin(self, epoch, logs=None):
+        self.epoch = epoch
+        progress_text.value = "Training epoch " + str(epoch + 1)
+        pass
+
+    def on_epoch_end(self, epoch, logs=None):
+        epoch_text.value = "Epoch " + str(epoch + 1) + " val_acc: " + str(logs.get('val_acc'))
+        pass
+
+    def on_train_batch_begin(self, batch, logs=None):
+        #print("***" + str(self.epoch) + str(self.train_steps_per_epoch) + str(batch))
+        pass
+
+    def on_train_batch_end(self, batch, logs=None):
+        progress_bar.value = self.epoch * self.train_steps_per_epoch + batch
+        if logs:
+            progress_text.value = 'Training epoch ' + str(self.epoch + 1) + ' loss: ' + str(
+                logs.get('loss'))
+
+    def on_test_begin(self, logs=None):
+        progress_text.value = "Validating epoch " + str(self.epoch + 1)
+        pass
+
+    def on_test_batch_begin(self, batch, logs=None):
+        pass
+
+    def on_test_batch_end(self, batch, logs=None):
+        pass
+
+    def on_predict_begin(self, logs=None):
+        progress_text.value = "Classifying"
+        progress_bar.max = demo.test_generator.n // demo.batch_size
+        pass
+
+    def on_predict_batch_begin(self, batch, logs=None):
+        pass
+
+    def on_predict_batch_end(self, batch, logs=None):
+        progress_bar.value = batch
+        pass
+
+    def on_predict_end(self, batch, logs=None):
+        progress_text.value = "Classification Complete"
+        pass
+
+
+demo = TransferLearningDemo.Demo()
+progress_info = ProgressBar(demo)
```

Comment:
Global vars are hard to maintain because they are hard to track the state of them. Are they really necessary? 

Also I don't see any `def main()` in this file, so I can only assume that these objects will be initialized whenever the module is imported. This might be confusing and lead to hard to track down initialization order bugs.